beard_server/tasks.py

```diff
@@ -22,19 +22,81 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-from __future__ import (
-    absolute_import,
-    division,
-    print_function)
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
 
 from .celery import app
-from .modules.clustering.beard import predict as clusters
+
+from .modules.clustering.beard import predict as make_beard_clusters
+from .modules.clustering.recid import make_recid_clusters
+from .modules.clustering.resolver import solve_claims_conflict
+from .modules.matching import match_clusters
 
 
 @app.task
 def make_clusters(records, signatures):
+    """Make clusters and match them with the current state of database.
+
+    This method allows for dispatching Celery tasks, that create
+    two sets of clusters, one created by the signatures having the same
+    recid and the second one using Beard library.
+
+    After creating two types of clusters, the clusters are matched with
+    each other, to link an output of Beard with the current state of the
+    database. The further logic of resolving conflicts is not the subject
+    of this module and is proceed by the client (invenio-beard).
+    """
+
+    # Create clusters using Beard.
+    try:
+        beard_clusters = make_beard_clusters(records, signatures)
+    except IOError:  # pragma: no cover
+        # Probably the file does not exist, ie. the model was not trained.
+        return "ERR: linkage.dat file not found. Train the model."
```

Comment:
I think it's better to re-raise an exception here. Returning a string is not compatible with the other return type: `(clusters_with_profiles, clusters_with_no_profiles)`